### PR TITLE
clang-tidy: Enable readability checks

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

**Comment by axic** [comment]
@chfast you want to land this?

File: test/utils/asserts.hpp [hunk 1]
```diff
@@ -8,7 +8,7 @@
 #include <gmock/gmock.h>
 #include <iosfwd>
 
-MATCHER(Traps, "")
+MATCHER(Traps, "")  // NOLINT(readability-redundant-string-init)
```
Comment (axic) [inline test/utils/asserts.hpp:11]: Btw what is this error?

File: test/utils/wabt_engine.cpp [hunk 2]
```diff
@@ -87,10 +87,7 @@ bool WabtEngine::instantiate(bytes_view wasm_binary)
 
     // Run start function (this will return ok if no start function is present)
     const wabt::interp::ExecResult r = m_executor.RunStartFunction(m_module);
-    if (r.result != wabt::interp::Result::Ok)
-        return false;
-
-    return true;
+    return r.result == wabt::interp::Result::Ok;
```
Comment (axic) [inline test/utils/wabt_engine.cpp:90]: I still hoped to prove both branches are covered with #381, but will add back the inverse of this commit for a test.

File: test/utils/asserts.hpp (hunk 1 above)
Comment (chfast) [reply test/utils/asserts.hpp:11]: It generates code like `std::string s = ""`.

**Pull request merged**